pkg/vault/contracts/RouterCommon.sol

```diff
@@ -191,8 +213,11 @@ abstract contract RouterCommon is IRouterCommon, VaultGuard {
      * returned ETH.
      */
     function _returnEth(address sender) internal {
-        uint256 excess = address(this).balance;
+        if (_isReturnEthLockedSlot().tload()) {
```

Comment:
Let's add a small comment explaining why we need this.
Something along the line `If the slot is locked, this is part of a larger operation and ETH needs to be returned at the very end.`